conmon-rs/server/src/child_reaper.rs

```diff
@@ -204,15 +209,20 @@ impl ReapableChild {
 
     fn watch(&self) -> Result<(Sender<ExitChannelData>, Receiver<ExitChannelData>)> {
         let exit_paths = self.exit_paths().clone();
+        let oom_exit_paths = self.oom_exit_paths().clone();
         let pid = self.pid();
         // Only one exit code will be written.
         let (exit_tx, exit_rx) = broadcast::channel(1);
         let exit_tx_clone = exit_tx.clone();
+        let mut cancel_rx = exit_tx.subscribe();
         let timeout = *self.timeout();
 
         task::spawn(async move {
             let exit_code: i32;
             let mut timed_out = false;
+            task::spawn(async move {
+                OOMWatcher::wait_for_oom(&mut cancel_rx, pid, &oom_exit_paths).await
```

Comment:
I think we should log the error here (`if let Err(e) = OOMWatcher::wait_for_oom(…) { error!("Unable to …: {}", e) }` otherwise we will lose the information if something went wrong.

Reply:
fixed